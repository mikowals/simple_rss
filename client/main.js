var DAY = 1000 * 60 * 60 * 24;
var articlesOnLoad = 20;
var updateNowFreq = 1 * 60 * 1000;

var intervalProcesses = []; //hold interval process id to start and stop with different functions.
Session.setDefault("loaded", false);
Session.setDefault("importOPML", false);
Session.setDefault( "now", new Date() );
Session.setDefault("articleLimit", articlesOnLoad );

var cleanForXml = function ( string ){
  string = string.replace ( /\"/g, "&quot;");
  string = string.replace ( /\'/g, "&apos;");
  string = string.replace ( /\&/g, "&amp;");
  string = string.replace ( /\>/g, "&gt;");
  
  return  string.replace ( /\</g, "&lt;");
};
                   
var articleSub;

Deps.autorun( function(){
  console.log( Session.get( "articleLimit" ) );
  articleSub = Meteor.subscribe( "feedsWithArticles", + Session.get( "articleLimit" ) );
});

Deps.autorun( function(){
  if ( articleSub && articleSub.ready()) 
    Session.set( "loaded",  true);
});

Meteor.startup( function() {
                          
  intervalProcesses['updateNow'] = intervalProcesses['updateNow'] || Meteor.setInterval( function() {
    Session.set( "now", new Date() );
    },
    updateNowFreq );
  Session.set( "offline", null);

});

Deps.autorun( function(){
    if ( ! Meteor.status().connected && Session.equals( "loaded", true) ) {
      console.log( "Meteor.status().connected = " + Meteor.status().connected );
      Session.set ("offline", "offline" );
    }
    else if ( articleSub && articleSub.ready() ){
      Session.set("offline", null);
    }
});


var timeago = function( some_date ){
  var now = new Date( Session.get( "now" ) );
  var referenceDate = new Date( some_date );

  var timeago = ( now -  referenceDate ) / DAY;

  if (Math.floor(timeago )  >= 2) return Math.floor(timeago ) + " days ago";
  else if (Math.floor(timeago )  >= 1) return Math.floor(timeago ) + " day ago";
  else if (Math.floor(timeago  * 24)  >= 2 ) return Math.floor(timeago * 24) + " hours ago";
  else if (Math.floor(timeago * 24)  >= 1 ) return Math.floor(timeago  * 24) + " hour ago";
  else if (Math.floor(timeago  * 24 * 60) >= 2) return Math.floor(timeago * 24 * 60) + " minutes ago";
  else {
    return "about a minute ago";
  }
};

Handlebars.registerHelper('timeago',  function(some_date){
  return timeago(some_date);
});

Template.feedList.feeds= function () {
  return Feeds.find({}, {sort: {title: 1}});
};

Template.feedList.flash = function(){
  return Session.get("feedListFlash");
};

Template.modalButtons.importOPML = function(){
  return Session.equals("importOPML", true);
};

Template.modalButtons.events({
                             
                             //could modify this to verify feed and populate fields for insertion
                             'submit, click #addFeed': function(e) {
                               e.stopImmediatePropagation();
                               e.preventDefault();
                               var url = $("#feedUrl").val();
                               var regex =/(((f|ht){1}tp|tps:\/\/)[-a-zA-Z0-9@:%_\+.~#?&\/\/=]+)/g;
                               if ( Match.test ( url, String ) && regex.test(url) ){
                                 Meteor.call( 'addFeed', { url: url } );
                                 $("#feedUrl").val("");
                               } else{
                                 alert("RSS feed entered is not a valid url");
                               }
                               return false;        
                             },
                             
                             'click #importToggle': function(){
                             Session.set("importOPML", true);
                             console.log(Session.equals("importOPML", true));
                             },
                             
                             'click #opmlUpload' : function(){
                               Session.set("importOPML", false);
                             
                               var opmlFile = $("#opmlFile")[0].files[0];
                               var fr = new FileReader();
                               fr.readAsText(opmlFile);
                               fr.onloadend = function(evt) {
                                 if (evt.target.readyState === FileReader.DONE) { // DONE == 2
                                   //Meteor.call( 'XML2JSparse', evt.target.result, function ( error, result ){
                                   $(  evt.target.result ).find( 'outline').each( function( ){
                                      var url = $( this ).attr("xmlUrl");
                                      if ( url )
                                        Meteor.call('addFeed', {url: url} );
                                   });
                                 }
                               }
                             },
                          
                             'click #importCancel' : function(){
                             Session.set("importOPML", false);
                             },
                             
                             'click #exportOPML': function(){
                               var exportOPML = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" +
                                               "<opml version=\"1.0\">" +
                                                "<head>" +
                                                 "<title>" + Meteor.user.username + " subscriptions from New-River</title>" +
                                                 "</head>" +
                                               "<body>";
                               Feeds.find().forEach( function( feed ) {
                                 exportOPML += "<outline " +
                                               "text=\"" + cleanForXml( feed.title ) + "\" " +
                                               "title=\"" + cleanForXml( feed.title ) + "\" " +
                                               "type=\"rss\" " +
                                               "xmlUrl=\"" + cleanForXml( feed.url ) + "\"/>";
                               });
                                exportOPML += "</body></opml>";
                                //exportOPML = ( new window.DOMParser() ).parseFromString( exportOPML, "text/xml");
                                //exportOPML = (new XMLSerializer()).serializeToString( exportOPML );
                                var blob = new Blob([exportOPML], {type: "application/xml"});
                                var fname = Meteor.absoluteUrl().split( "//" )[1];
                                saveAs ( blob, fname + ".opml");
                             }
                         
                             });

Template.feedList.events({                                                   
                         'click #removeFeed': function(){
                           Meteor.call( 'removeFeed', Session.get("selected_feed"), function( error ){
                             if ( error === false ) console.error( "could not remove feed"); 
                             else Session.set( "selected_feed", null);
                             });
                         }
                         
                         });
                           
Template.articleList.articles = function() {
  
   return Articles.find( {}, { sort: { date: -1, _id: 1} } );
};

Template.articleList.loaded = function(){
  return Session.equals( "loaded", true );
};

Template.articleList.events({
  'click a, contextmenu a': function( e ){
     if ( ! Session.equals( "handleTap", true)){
       Meteor.call( 'markRead', $( e.currentTarget ).attr('href'));
     } else{
       e.preventDefault();
       e.stopImmediatePropagation();
       return false;
     }
     return true;
  },
  
  'tap a':  function( e ){
    Session.set( "handleTap", true);
    e.preventDefault();
    e.stopImmediatePropagation();
    var dest = $( e.currentTarget ).attr('href');
    Meteor.call( 'markRead', dest, function(){
      window.location.href = dest;
    });
    return false;
  }
}); 

Template.article.subscribed = function(){
  return Feeds.findOne({title: this.source}) !==null;
};

Template.article.subscribeRss = function(){
  if (this.sourceUrl) return this.sourceUrl;

  var feed = Feeds.findOne( {title: this.source});
  return feed && feed.url;
  
};

Template.menubar.loaded = function(){
  return Session.equals( "loaded", true );
};

Template.menubar.offline = function(){
  return Session.get("offline");
}

Template.updated.updated = function(){
  return Session.get( "updated" );
}

Template.feed.selected = function () {
  return Session.equals("selected_feed", this._id) ? "selected" : '';
};

Template.feed.article_count = function () {
  return Articles.find({source: this.title}).count();
};

Template.feed.events({
                     'click': function(){
                     Session.set("selected_feed", this._id);
                     }
                     });


Template.articleList.rendered = function(){

  $('#footer').waypoint( function( evt, dir) {
    if ( dir === 'down' )
      Session.set( "articleLimit", Session.get( "articleLimit" ) + 20 );
    },
    { offset: '110%'}); 
};
