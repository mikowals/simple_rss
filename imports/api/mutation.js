import { gql } from '@apollo/client';

export const REMOVE_FEED = gql`
  mutation removeFeed($id: String) {
    removeFeed(id: $id) {
      _id
    }
  }
`;

export const ADD_FEED = gql`
mutation addFeed($url: String) {
  addFeed(url: $url) {
    _id
    title
    last_date
    url
    count
  }
}
`
