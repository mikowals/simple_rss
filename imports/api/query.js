import { gql } from '@apollo/client';

export const ARTICLES_QUERY = gql`
  query($userId: String!) {
    articles(userId: $userId) {
      _id
      source
      date
      title
      link
      summary
      feed_id
    }
  }
`;

export const FEEDS_QUERY = gql`
  query feeds($userId: String!) {
    feeds(userId: $userId) {
        _id
        title
        url
        last_date
        count
    }
  }
`;

export const FEED_COUNT = gql`
  query articlesCountByFeed($id: String) {
    articlesCountByFeed(id: $id) {
      _id
      count
    }
  }
`;

export const FEED_IDS = gql`
  query feedIds($userId: String!) {
    feedIds(userId: $userId)
  }
`

export const USER_QUERY = gql`
  query user($userId: String!) {
    user(userId: $userId) {
      _id
      feeds {
        _id
        title
        url
        last_date
        count
      }
    }
  }
`
