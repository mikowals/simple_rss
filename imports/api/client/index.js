import React from 'react';
import { hydrateRoot } from 'react-dom';
import { BrowserRouter, Route, Routes, Link } from 'react-router-dom';
import { Meteor } from 'meteor/meteor';
import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';
import { BatchHttpLink } from "@apollo/client/link/batch-http";
import { FeedsPageWithContainer } from '/imports/ui/feeds';
import { ArticlesPageWithStream } from '/imports/ui/articles';

export const client = new ApolloClient({
  cache: new InMemoryCache().restore(window.__APOLLO_STATE__),
  link: new BatchHttpLink({
    uri: '/graphql',
    credentials: 'same-origin',
    batchInterval: 10}),
  connectToDevTools: true
});

export const renderRoutes = () => (
    <BrowserRouter>
      <ApolloProvider client={client}>
        <Routes>
          <Route exact path="/feeds" element={<FeedsPageWithContainer />} />
          <Route exact path="/articles" element={<ArticlesPageWithStream />} />
          <Route element={<ArticlesPageWithStream />} />
        </Routes>
      </ApolloProvider>
      <Link to="/feeds" >Feeds</Link>
      <Link to="/articles" >Articles</Link>
    </BrowserRouter>
);


Meteor.startup(() => {
  hydrateRoot(
    document.getElementById('app'),
    renderRoutes()
  )
});
