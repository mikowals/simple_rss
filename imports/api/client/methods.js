import { Meteor } from 'meteor/meteor';
import { Articles, Feeds } from '/imports/api/simple_rss';

var handleError = function( error ){
  if ( error ) console.log( error );
};

Meteor.methods({
  removeFeed: function (feedId) {
    var self = this;
    var userId = self.userId || 'nullUser';
    check(feedId, String);

    var feed = Feeds.findOne( {_id: feedId, subscribers: userId}, {fields:{_id: 1, subscribers: 1, url: 1, hub: 1}});

    Articles.remove({feed_id: feedId});
    Feeds.remove(feedId);
  },

  addFeed: function( doc ){
    var self = this;
    self.unblock();
    check ( doc, { url: String});
    var originalUrl = doc.url;
    var userId = self.userId || 'nullUser';

    // create id so we can insert articles before feed itself is inserted.
    doc._id = Feeds._makeNewID();
    if ( self.isSimulation ) {
      Feeds.insert( doc, handleError );
      return;
    }
  }
});
