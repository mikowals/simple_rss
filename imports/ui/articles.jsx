import { Meteor } from 'meteor/meteor';
import { withTracker } from 'meteor/react-meteor-data';
import React, { useState, useEffect, memo } from 'react';
import PropTypes from 'prop-types';
import { Articles } from '/imports/api/simple_rss';
import { withTimeText, TimeAgo } from './timeAgo';
import { initialArticleLimit } from '/imports/api/simple_rss';

// Spread the article object (...) to avoid new object causing rerender.
const renderArticle = (article) => {
  return <ArticleWithTimeAgo {...article} key={article._id} />;
};

const articlesEqual = (prev, next) => {
  if (prev.articles.length !== next.articles.length) return false;
  for (let ii = 0; ii < prev.articles.length; ii++) {
    const prevValues = Object.values(prev.articles[ii]);
    const nextValues = Object.values(next.articles[ii]);
    for (let jj = 0; jj < prevValues.length; jj++){
      if (prevValues[jj] !== nextValues[jj]) {
        console.log("Articles not equal ", ii, prevValues, nextValues);
        return false;
      }
    }
  }
  return true;
};

export const ArticlesPage = memo((props) => {
  return <div id="stream">{props.articles.map(renderArticle)}</div>
}, articlesEqual);

ArticlesPage.propTypes = {
  articles: PropTypes.array
};

ArticlesPage.displayName = "ArticlesPage";

export const ArticlesPageContainer = withTracker(() => {
    let limit = Session.get('articleLimit')
    var list = Articles.find({}, {limit: limit, sort: {date: -1, _id: 1}});
    return {
      articles: list.map((article) => {
        // Convert date so simple equality checks work and avoid rerender.
        article.date = new Date(article.date).getTime();
        // Some articles don't have titles and display nicer with placeholder.
        article.title = article.title || "Link";
        return article;
      })
    };
})(ArticlesPage);

ArticlesPageContainer.displayName = "ArticlesPageContainer";

const Article = (props) => {
    var {_id, title, source, summary, link, timeText} = props;
    // TimeAgo occurs twice.  Once in SourceHeader and once in the footer.
    return  <div id={_id} className="section">
              <SourceHeader source={source} timeText={timeText} />
              <div className="article">
                <TitleAndSummary link={link} title={title} summary={summary} />
                <div className="footer visible-xs">
                  <time><TimeAgo timeText={timeText} /></time>
                </div>
              </div>
            </div>;
};

Article.propTypes = {
  _id: PropTypes.string.isRequired,
  timeText: PropTypes.string.isRequired,
  title: PropTypes.string,
  source: PropTypes.string.isRequired,
  summary: PropTypes.string,
  link: PropTypes.string.isRequired,
};

const ArticleWithTimeAgo = withTimeText(Article);

const SourceHeader = (props) => {
    const {source, timeText} = props;
    return <div className="header row-fluid" name="source">
             <h2>{source}</h2>
             <span className="spacer"/>
             <time><TimeAgo timeText={timeText} /></time>
           </div>;
};

const TitleAndSummary = React.memo((props) => {
    const {link, title, summary} = props;
    return <React.Fragment>
           <div className="header">
             <h3>
               <a href={link}>{title}</a>
            </h3>
          </div>
          <div className="description"
               dangerouslySetInnerHTML={{__html: UniHTML.purify(summary)}}/>
          </React.Fragment>;
});

TitleAndSummary.displayName = "TitleAndSummary";

//ArticleWithTimeAgo.displayName = "ArticleWithTimeAgo";
