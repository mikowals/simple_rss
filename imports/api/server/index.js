import { ApolloServer, gql } from 'apollo-server-express';
import { WebApp } from 'meteor/webapp';
import { Feeds, Articles } from '/imports/api/simple_rss';
import React from 'react';
import { renderToStaticMarkup, renderToNodeStream, renderToString } from 'react-dom/server';
import { ApolloClient, InMemoryCache, ApolloProvider, createHttpLink} from '@apollo/client';
import { renderToStringWithData } from "@apollo/client/react/ssr";
import fetch from 'cross-fetch';
import { ApolloApp } from '/imports/ui/app';
import { FeedsPage } from '/imports/ui/feeds';
import { ArticlesPage } from '/imports/ui/articles'
import { ServerStyleSheet } from "styled-components";
import { Meteor } from 'meteor/meteor';
import { onPageLoad } from 'meteor/server-render';
import { resolvers } from '/imports/api/server/resolvers';
import { typeDefs } from '/imports/api/server/typeDefs';
//import { WebApp } from 'meteor/webapp'
//import { makeExecutableSchema } from 'graphql-tools';

const server = new ApolloServer({typeDefs, resolvers});

server.applyMiddleware({
  app: WebApp.connectHandlers,
  path: '/graphql'
})

BrowserPolicy.content.allowOriginForAll("https://fonts.googleapis.com");
BrowserPolicy.content.allowOriginForAll("http://cdn.jsdelivr.net");
BrowserPolicy.content.allowOriginForAll("http://fonts.gstatic.com");
// We are doing this work-around because Playground sets headers and WebApp also sets headers
// Resulting into a conflict and a server side exception of "Headers already sent"
WebApp.connectHandlers.use('/graphql', (req, res) => {
  if (req.method === 'GET') {
    res.end()
  }
});

// Wrap ArticlesPage here so that we can inject it into div id='stream'.
const SSRPage = ({client, location}) => (
  <ApolloProvider client={client}>
    {location === "/feeds" ? <FeedsPage /> : <ArticlesPage />}
  </ApolloProvider>
);

function Html({ content, state }) {
  return (
    <html>
      <body>
        <div id="app">
        <div id="stream" dangerouslySetInnerHTML={{ __html: content }} />
        </div>
        <script dangerouslySetInnerHTML={{
          __html: `window.__APOLLO_STATE__=${JSON.stringify(state).replace(/</g, '\\u003c')};`,
        }} />

      </body>
    </html>
  );
}

function AppWithCache({ content, state, location }) {
  return (
    <>
    {<div id={location === "/feeds" ? "feed-container" : "stream"}
      dangerouslySetInnerHTML={{ __html: content }} />}
    <script dangerouslySetInnerHTML={{
      __html: `window.__APOLLO_STATE__=${JSON.stringify(state).replace(/</g, '\\u003c')};`,
    }} />
    </>
  );
}

// this is faster than server-render 'onPageLoad' but lacks css and js to continue updates.
WebApp.connectHandlers.use('/static', (req, res, next) => {
  const client = new ApolloClient({
    ssrMode: true,
    // Remember that this is the interface the SSR server will use to connect to the
    // API server, so we need to ensure it isn't firewalled, etc
    link: createHttpLink({
      uri: 'http://localhost:3000/graphql',
      credentials: 'same-origin',
      fetch
    }),
    cache: new InMemoryCache(),
  });
  renderToStringWithData(SSRPage({client: client})).then((content) => {
    const initialState = client.extract();
    const html = <Html content={content} state={initialState} />;
    res.writeHead(
      200,
      {'Content-Type': 'text/html'}
    );
    renderToNodeStream(html).pipe(res)
  })
});

onPageLoad(async sink => {
  const client = new ApolloClient({
    ssrMode: true,
    // Remember that this is the interface the SSR server will use to connect to the
    // API server, so we need to ensure it isn't firewalled, etc
    link: createHttpLink({
      uri: 'http://localhost:3000/graphql',
      credentials: 'same-origin',
      fetch
    }),
    cache: new InMemoryCache(),
  });

  const sheet = new ServerStyleSheet();
  const location = sink.request.url.pathname;
  const content = await renderToStringWithData(SSRPage({client, location}));

  const initialState = client.extract();
  const appJSX = <AppWithCache content={content} state={initialState} location={location} />;

  const htmlStream = renderToNodeStream(appJSX);
  sink.renderIntoElementById("app", htmlStream);

});
