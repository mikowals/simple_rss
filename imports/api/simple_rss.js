import { Mongo } from 'meteor/mongo';
import { Meteor } from 'meteor/meteor';
import { onPageLoad } from 'meteor/server-render';
import { renderToNodeStream, renderToString } from 'react-dom/server';
import React from 'react';
import { unstable_createRoot } from 'react-dom';
import { ArticlesPage } from '/imports/ui/articles';
import { AppContainer, App } from '/imports/ui/app';

_ = lodash;

export const Feeds = new Mongo.Collection ( "feeds" );
export const Articles = new Mongo.Collection ( "articles" );
export const initialArticleLimit = 40;
export const DAY = 1000 * 60 * 60 * 24;
export const keepLimitDate = function(){
  return new Date( new Date().getTime() - ( DAY * daysStoreArticles ));
};
export const daysStoreArticles = 3.0;

if (Meteor.isServer) {
  BrowserPolicy.content.allowConnectOrigin("*.mak-play.com");
  BrowserPolicy.content.allowEval();

  Feeds._ensureIndex( { url: 1 }, {unique: true} );
  Articles._ensureIndex( { link: 1 }, {unique: true, dropDups: true });
  Articles._ensureIndex( { feed_id: 1, date: -1} );

  //Accounts.config({sendVerificationEmail: true});
  Meteor.users.deny({
    update: () => true
  });

  onPageLoad(sink => {
    //need to rewrite this for real userID using cookies;
    const userId = "nullUser";
    const feedList = Meteor.users.findOne(
      {_id: userId},
      {fields: {feedList: 1}}
    ).feedList;
    const articlesCursor = Articles.find(
      {feed_id: {$in: feedList}},
      {limit: initialArticleLimit, sort: {date: -1, _id: 1}}
    );
    const articles = articlesCursor.fetch();

    const htmlStream = renderToNodeStream(<ArticlesPage articlesFromServer={articles}/>);
    sink.renderIntoElementById("app", htmlStream);
  });
} else {
  onPageLoad( sink => {
    Meteor.subscribe('articles', function(){
      const rootElement = document.getElementById("app");
      unstable_createRoot(rootElement, {
        hydrate: true
      }).render(<App />);
    });
  });
}
