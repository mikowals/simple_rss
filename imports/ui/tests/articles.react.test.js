import React from 'react';
import { Article, ArticlesPage } from '../articles';
import { ARTICLES_QUERY } from '../../api/query';
import {cleanup, MockedProvider} from '@apollo/client/testing';
import { InMemoryCache } from '@apollo/client';
import renderer, {act} from 'react-test-renderer';
import "babel-polyfill";
import wait from 'waait';

const article1 = {
  // date equal to now necessary for text to always match.
  date: (new Date()).getTime(),
  source: "aPublisher",
  summary: "blah blah blah",
  link: "http://test.example.com/1",
  title: "An Interesting Article",
  _id: "abc123",
  feed_id: "feed1",
}

const article2 = {
  // date equal to now necessary for text to always match.
  date: (new Date()).getTime() - 5000,
  source: "unknown",
  summary: "ipso facto",
  link: "http://ipso.com/1",
  title: "An Boring Article",
  _id: "def456",
  feed_id: "feed2",
}

afterEach(() => cleanup);

it('renders accurately', () => {
  const props = article1;

  const component = renderer.create(
    <Article {...props} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a list of articles', async () => {
  const mocks = [
    {
      request: {
        query: ARTICLES_QUERY,
        variables: {userId: "nullUser"},
      },
      result: {
        data: {
          articles: [article1, article2]
        }
      }
    }
  ]

  const component = renderer.create(
    <MockedProvider mocks={mocks} addTypename={false} >
      <ArticlesPage />
    </MockedProvider>
  );
  expect(component.root.findAllByType('li')).toHaveLength(0);
  await act(wait);
  expect(component.root.findAllByType('li')).toHaveLength(2);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
