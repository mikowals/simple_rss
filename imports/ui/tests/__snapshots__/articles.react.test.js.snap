// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of articles 1`] = `
Array [
  <li
    className="section"
    id="abc123"
  >
    <div
      className="header row-fluid"
      name="source"
    >
      <h2>
        aPublisher
      </h2>
      <span
        className="spacer"
      />
      <time>
        <span>
          less than a minute ago
        </span>
      </time>
    </div>
    <div
      className="article"
    >
      <div
        className="header"
      >
        <h3>
          <a
            href="http://test.example.com/1"
          >
            An Interesting Article
          </a>
        </h3>
      </div>
      <div
        className="description"
        dangerouslySetInnerHTML={
          Object {
            "__html": "blah blah blah",
          }
        }
      />
      <div
        className="footer visible-xs"
      >
        <time>
          <span>
            less than a minute ago
          </span>
        </time>
      </div>
    </div>
  </li>,
  <li
    className="section"
    id="def456"
  >
    <div
      className="header row-fluid"
      name="source"
    >
      <h2>
        unknown
      </h2>
      <span
        className="spacer"
      />
      <time>
        <span>
          less than a minute ago
        </span>
      </time>
    </div>
    <div
      className="article"
    >
      <div
        className="header"
      >
        <h3>
          <a
            href="http://ipso.com/1"
          >
            An Boring Article
          </a>
        </h3>
      </div>
      <div
        className="description"
        dangerouslySetInnerHTML={
          Object {
            "__html": "ipso facto",
          }
        }
      />
      <div
        className="footer visible-xs"
      >
        <time>
          <span>
            less than a minute ago
          </span>
        </time>
      </div>
    </div>
  </li>,
]
`;

exports[`renders accurately 1`] = `
<li
  className="section"
  id="abc123"
>
  <div
    className="header row-fluid"
    name="source"
  >
    <h2>
      aPublisher
    </h2>
    <span
      className="spacer"
    />
    <time>
      <span>
        less than a minute ago
      </span>
    </time>
  </div>
  <div
    className="article"
  >
    <div
      className="header"
    >
      <h3>
        <a
          href="http://test.example.com/1"
        >
          An Interesting Article
        </a>
      </h3>
    </div>
    <div
      className="description"
      dangerouslySetInnerHTML={
        Object {
          "__html": "blah blah blah",
        }
      }
    />
    <div
      className="footer visible-xs"
    >
      <time>
        <span>
          less than a minute ago
        </span>
      </time>
    </div>
  </div>
</li>
`;
